### PR TITLE
TILA-2125 dynamic form validation

### PR DESCRIPTION
**Pull request by kieferbonk**

### CODE REVIEW

File: ui/components/reservation/Step0.tsx [hunk 1]
```diff
@@ -130,7 +148,48 @@ const Step0 = ({
...
+                : reserveeType?.toLocaleLowerCase() || "individual";
+              return (
+                <li>
+                  <ErrorAnchor
+                    href="javascript:void(0);"
+                    onClick={() => {
+                      const element =
+                        document.getElementById(key) ||
+                        document.getElementById(`${key}-label`);
+                      const top = element.getBoundingClientRect()?.y;
```
Comment (joonatank) [inline ui/components/reservation/Step0.tsx:173]: element: Element | null

File: ui/components/reservation/Step0.tsx [hunk 2]
```diff
@@ -130,7 +148,48 @@ const Step0 = ({
...
+                      window.scroll({
+                        top: window.scrollY + top - 28,
```
Comment (joonatank) [inline ui/components/reservation/Step0.tsx:175]: top: number | null

File: ui/components/reservation/Step1.tsx [hunk 3]
```diff
@@ -56,6 +65,19 @@ const PreviewValue = styled.span`
   font-size: var(--fontsize-body-l);
 `;
 
+const scrollToBox = (id: string): void => {
+  const element = document.getElementById(id);
+  const checkbox = document.getElementById(`${id}-terms-accepted`);
+
+  const top = element.getBoundingClientRect()?.y;
+  window.scroll({
+    top: window.scrollY + top - 28,
```
Comment (joonatank) [inline ui/components/reservation/Step1.tsx:74]: top: number | null

File: ui/components/reservation/Step1.tsx [hunk 4]
```diff
@@ -56,6 +65,19 @@ const PreviewValue = styled.span`
   font-size: var(--fontsize-body-l);
 `;
 
+const scrollToBox = (id: string): void => {
+  const element = document.getElementById(id);
+  const checkbox = document.getElementById(`${id}-terms-accepted`);
```
Comment (joonatank) [inline ui/components/reservation/Step1.tsx:70]: getByElementId: (...) => Element | null

**Review by joonatank — CHANGES_REQUESTED** [review]
Null checks.

File: common/src/reservation-form/ReservationFormField.tsx [hunk 5]
```diff
@@ -23,7 +23,10 @@ type Props = {
   required: boolean;
   // Not good to pass the translation function here but this is because this is shared between ui and admin
   // and admin is lacking translation namespaces
-  t: (key: string) => string;
+  t: (
+    key: string,
+    options?: Record<string, string | number | undefined>
+  ) => string;
```
[the remark this answers is not in the document]
Comment (kieferbonk) [reply common/src/reservation-form/ReservationFormField.tsx:27]: Tried that, but became too hairy on admin side, which also uses the components.

**Review by joonatank — APPROVED** [review]

**Pull request merged**